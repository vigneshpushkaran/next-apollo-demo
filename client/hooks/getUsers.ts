import { useQuery, gql } from "@apollo/client";

export const GET_USERS = gql`
  query getUsers($cursor: ID) {
    users(first: 20, cursor: $cursor) {
      edges {
        node {
            id,
            name
            title
            avatar
            empId
            jobTitle
            jobDesc
            companyName
        }
      }
      pageInfo {
        endCursor
        hasNextPage
      }
    }
  }
`;

function useUsers() {
  const { data, loading, fetchMore } = useQuery(GET_USERS, {
    notifyOnNetworkStatusChange: true,
  });
  if (loading && !data) {
    console.log("data1 ", data)
    return { loading, users: [] };
  }
  const loadMore = () => {
    return fetchMore({
      query: GET_USERS,
      variables: {
        cursor: data.users.pageInfo.endCursor,
      },
      updateQuery: (previousResult, { fetchMoreResult }) => {
        const newEdges = fetchMoreResult.users.edges;
        const pageInfo = fetchMoreResult.users.pageInfo;
        return newEdges.length
          ? {
            users: {
              __typename: previousResult.users.__typename,
              edges: [...previousResult.users.edges, ...newEdges],
              pageInfo,
            },
          }
          : previousResult;
      },
    });
  };
  return {
    users: data.users.edges.map(({ node }) => node),
    hasNextPage: data.users.pageInfo.hasNextPage,
    loading,
    loadMore,
  };
}
export default useUsers;