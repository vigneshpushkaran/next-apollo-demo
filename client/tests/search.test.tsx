
import React from 'react';
import App from '../pages/search';
import { MockedProvider } from '@apollo/react-testing';
import { render, cleanup } from "@testing-library/react";
import { GET_USERS } from '../hooks/getUsers'

const mocks = [
    {
        request: {
            query: GET_USERS
        },
        result: {
            data: {
                "users": {
                    "edges": [
                        {
                            "node": {
                                "id": "ppaf8ojpph",
                                "name": "Willa Jast",
                                "title": "Mr.",
                                "avatar": "data:image/svg+xml;charset=UTF-8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20version%3D%221.1%22%20baseProfile%3D%22full%22%20width%3D%22undefined%22%20height%3D%22undefined%22%3E%20%3Crect%20width%3D%22100%25%22%20height%3D%22100%25%22%20fill%3D%22grey%22%2F%3E%20%20%3Ctext%20x%3D%220%22%20y%3D%2220%22%20font-size%3D%2220%22%20text-anchor%3D%22start%22%20fill%3D%22white%22%3Eundefinedxundefined%3C%2Ftext%3E%20%3C%2Fsvg%3E",
                                "empId": "1",
                                "jobTitle": "Corporate Markets Administrator",
                                "jobDesc": "Et qui quia odio et.\nEligendi in deserunt.\nHarum sit odio quia vitae provident quo provident.\nHarum ullam pariatur quos est quod.",
                                "companyName": "Hermiston - Reichert"
                            }
                        },
                        {
                            "node": {
                                "id": "h5f2bwfrwo",
                                "name": "Jillian Kohler",
                                "title": "Dr.",
                                "avatar": "data:image/svg+xml;charset=UTF-8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20version%3D%221.1%22%20baseProfile%3D%22full%22%20width%3D%22undefined%22%20height%3D%22undefined%22%3E%20%3Crect%20width%3D%22100%25%22%20height%3D%22100%25%22%20fill%3D%22grey%22%2F%3E%20%20%3Ctext%20x%3D%220%22%20y%3D%2220%22%20font-size%3D%2220%22%20text-anchor%3D%22start%22%20fill%3D%22white%22%3Eundefinedxundefined%3C%2Ftext%3E%20%3C%2Fsvg%3E",
                                "empId": "b",
                                "jobTitle": "Product Paradigm Analyst",
                                "jobDesc": "Nobis et sint magnam aut atque.\nMaiores incidunt occaecati iusto laborum facere.\nDeleniti dolorem quibusdam.\nVeniam perspiciatis et nisi aut corporis laboriosam.",
                                "companyName": "Pfeffer - Parisian"
                            }
                        },
                        {
                            "node": {
                                "id": "866hk73iv5",
                                "name": "Mitchel Ziemann",
                                "title": "Mrs.",
                                "avatar": "data:image/svg+xml;charset=UTF-8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20version%3D%221.1%22%20baseProfile%3D%22full%22%20width%3D%22undefined%22%20height%3D%22undefined%22%3E%20%3Crect%20width%3D%22100%25%22%20height%3D%22100%25%22%20fill%3D%22grey%22%2F%3E%20%20%3Ctext%20x%3D%220%22%20y%3D%2220%22%20font-size%3D%2220%22%20text-anchor%3D%22start%22%20fill%3D%22white%22%3Eundefinedxundefined%3C%2Ftext%3E%20%3C%2Fsvg%3E",
                                "empId": "r",
                                "jobTitle": "Product Infrastructure Analyst",
                                "jobDesc": "Minima sequi voluptas optio.\nVoluptatem excepturi eligendi.\nIste eius iure commodi molestiae at voluptas quo ex reprehenderit.\nIncidunt corporis vel.",
                                "companyName": "Lowe - Purdy"
                            }
                        },
                        {
                            "node": {
                                "id": "cvi1nj384v",
                                "name": "Maida Dickinson",
                                "title": "Mr.",
                                "avatar": "data:image/svg+xml;charset=UTF-8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20version%3D%221.1%22%20baseProfile%3D%22full%22%20width%3D%22undefined%22%20height%3D%22undefined%22%3E%20%3Crect%20width%3D%22100%25%22%20height%3D%22100%25%22%20fill%3D%22grey%22%2F%3E%20%20%3Ctext%20x%3D%220%22%20y%3D%2220%22%20font-size%3D%2220%22%20text-anchor%3D%22start%22%20fill%3D%22white%22%3Eundefinedxundefined%3C%2Ftext%3E%20%3C%2Fsvg%3E",
                                "empId": "z",
                                "jobTitle": "Forward Configuration Designer",
                                "jobDesc": "Cum ipsam consequatur ex accusamus ipsum.\nEum inventore laboriosam deleniti omnis occaecati.\nCulpa occaecati nemo alias et doloribus expedita.\nUt quia ut nostrum ducimus occaecati veniam ut exercitationem voluptatibus.",
                                "companyName": "Gerhold - Rau"
                            }
                        },
                        {
                            "node": {
                                "id": "2f07vgwj73",
                                "name": "Mrs. Reanna Walter",
                                "title": "Dr.",
                                "avatar": "data:image/svg+xml;charset=UTF-8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20version%3D%221.1%22%20baseProfile%3D%22full%22%20width%3D%22undefined%22%20height%3D%22undefined%22%3E%20%3Crect%20width%3D%22100%25%22%20height%3D%22100%25%22%20fill%3D%22grey%22%2F%3E%20%20%3Ctext%20x%3D%220%22%20y%3D%2220%22%20font-size%3D%2220%22%20text-anchor%3D%22start%22%20fill%3D%22white%22%3Eundefinedxundefined%3C%2Ftext%3E%20%3C%2Fsvg%3E",
                                "empId": "2",
                                "jobTitle": "Senior Mobility Architect",
                                "jobDesc": "Molestias ab esse impedit aliquam quo vel dolorem.\nDolor consequatur quis nobis nemo asperiores nam ea aut.\nNam corrupti necessitatibus quas.\nEa odio sit odio quae voluptas id error.",
                                "companyName": "Bogan Group"
                            }
                        }
                    ],
                    "pageInfo": {
                        "endCursor": "2f07vgwj73",
                        "hasNextPage": true
                    }
                }
            },
        },
    }
];

afterEach(cleanup);

test('renders without error with snapshot', async () => {
    const { findByText, asFragment, getByText } = render(
        <MockedProvider mocks={mocks} addTypename={false}>
            <App />
        </MockedProvider>,
    );
    expect(getByText("Loading...")).toBeInTheDocument();

    const buttonTag = await findByText("Home");
    expect(buttonTag).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();

    expect(getByText(/Willa Jast/i)).toBeInTheDocument();
});

