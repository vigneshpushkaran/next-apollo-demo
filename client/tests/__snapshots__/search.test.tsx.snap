// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without error with snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="layout"
    >
      <div
        class="makeStyles-root-1"
      >
        <header
          class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
          >
            <h6
              class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
            >
              Demo - PS
            </h6>
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
              href="/"
              tabindex="0"
            >
              <span
                class="MuiButton-label"
              >
                Home
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
              href="/search"
              tabindex="0"
            >
              <span
                class="MuiButton-label"
              >
                Search
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </div>
        </header>
      </div>
      <div
        style="margin: 8px 16px;"
      >
        <div>
          <div
            class="jsx-2990347280 cardParent"
          >
            <div
              class="MuiPaper-root MuiCard-root makeStyles-root-4 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                class="MuiCardHeader-root"
              >
                <div
                  class="MuiCardHeader-avatar"
                >
                  <div
                    aria-label="recipe"
                    class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-7"
                  >
                    <img
                      class="MuiAvatar-img"
                      srcset="data:image/svg+xml;charset=UTF-8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20version%3D%221.1%22%20baseProfile%3D%22full%22%20width%3D%22undefined%22%20height%3D%22undefined%22%3E%20%3Crect%20width%3D%22100%25%22%20height%3D%22100%25%22%20fill%3D%22grey%22%2F%3E%20%20%3Ctext%20x%3D%220%22%20y%3D%2220%22%20font-size%3D%2220%22%20text-anchor%3D%22start%22%20fill%3D%22white%22%3Eundefinedxundefined%3C%2Ftext%3E%20%3C%2Fsvg%3E"
                    />
                  </div>
                </div>
                <div
                  class="MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
                  >
                    Mr. Willa Jast
                  </span>
                  <span
                    class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                  >
                    Corporate Markets Administrator
                  </span>
                </div>
              </div>
              <div
                class="MuiCardContent-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Hermiston - Reichert
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                >
                  Et qui quia odio et.
Eligendi in deserunt.
Harum sit odio quia vitae provident quo provident.
Harum ullam pariatur quos est quod.
                </p>
              </div>
            </div>
            <div
              class="MuiPaper-root MuiCard-root makeStyles-root-4 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                class="MuiCardHeader-root"
              >
                <div
                  class="MuiCardHeader-avatar"
                >
                  <div
                    aria-label="recipe"
                    class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-7"
                  >
                    <img
                      class="MuiAvatar-img"
                      srcset="data:image/svg+xml;charset=UTF-8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20version%3D%221.1%22%20baseProfile%3D%22full%22%20width%3D%22undefined%22%20height%3D%22undefined%22%3E%20%3Crect%20width%3D%22100%25%22%20height%3D%22100%25%22%20fill%3D%22grey%22%2F%3E%20%20%3Ctext%20x%3D%220%22%20y%3D%2220%22%20font-size%3D%2220%22%20text-anchor%3D%22start%22%20fill%3D%22white%22%3Eundefinedxundefined%3C%2Ftext%3E%20%3C%2Fsvg%3E"
                    />
                  </div>
                </div>
                <div
                  class="MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
                  >
                    Dr. Jillian Kohler
                  </span>
                  <span
                    class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                  >
                    Product Paradigm Analyst
                  </span>
                </div>
              </div>
              <div
                class="MuiCardContent-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Pfeffer - Parisian
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                >
                  Nobis et sint magnam aut atque.
Maiores incidunt occaecati iusto laborum facere.
Deleniti dolorem quibusdam.
Veniam perspiciatis et nisi aut corporis laboriosam.
                </p>
              </div>
            </div>
            <div
              class="MuiPaper-root MuiCard-root makeStyles-root-4 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                class="MuiCardHeader-root"
              >
                <div
                  class="MuiCardHeader-avatar"
                >
                  <div
                    aria-label="recipe"
                    class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-7"
                  >
                    <img
                      class="MuiAvatar-img"
                      srcset="data:image/svg+xml;charset=UTF-8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20version%3D%221.1%22%20baseProfile%3D%22full%22%20width%3D%22undefined%22%20height%3D%22undefined%22%3E%20%3Crect%20width%3D%22100%25%22%20height%3D%22100%25%22%20fill%3D%22grey%22%2F%3E%20%20%3Ctext%20x%3D%220%22%20y%3D%2220%22%20font-size%3D%2220%22%20text-anchor%3D%22start%22%20fill%3D%22white%22%3Eundefinedxundefined%3C%2Ftext%3E%20%3C%2Fsvg%3E"
                    />
                  </div>
                </div>
                <div
                  class="MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
                  >
                    Mrs. Mitchel Ziemann
                  </span>
                  <span
                    class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                  >
                    Product Infrastructure Analyst
                  </span>
                </div>
              </div>
              <div
                class="MuiCardContent-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Lowe - Purdy
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                >
                  Minima sequi voluptas optio.
Voluptatem excepturi eligendi.
Iste eius iure commodi molestiae at voluptas quo ex reprehenderit.
Incidunt corporis vel.
                </p>
              </div>
            </div>
            <div
              class="MuiPaper-root MuiCard-root makeStyles-root-4 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                class="MuiCardHeader-root"
              >
                <div
                  class="MuiCardHeader-avatar"
                >
                  <div
                    aria-label="recipe"
                    class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-7"
                  >
                    <img
                      class="MuiAvatar-img"
                      srcset="data:image/svg+xml;charset=UTF-8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20version%3D%221.1%22%20baseProfile%3D%22full%22%20width%3D%22undefined%22%20height%3D%22undefined%22%3E%20%3Crect%20width%3D%22100%25%22%20height%3D%22100%25%22%20fill%3D%22grey%22%2F%3E%20%20%3Ctext%20x%3D%220%22%20y%3D%2220%22%20font-size%3D%2220%22%20text-anchor%3D%22start%22%20fill%3D%22white%22%3Eundefinedxundefined%3C%2Ftext%3E%20%3C%2Fsvg%3E"
                    />
                  </div>
                </div>
                <div
                  class="MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
                  >
                    Mr. Maida Dickinson
                  </span>
                  <span
                    class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                  >
                    Forward Configuration Designer
                  </span>
                </div>
              </div>
              <div
                class="MuiCardContent-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Gerhold - Rau
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                >
                  Cum ipsam consequatur ex accusamus ipsum.
Eum inventore laboriosam deleniti omnis occaecati.
Culpa occaecati nemo alias et doloribus expedita.
Ut quia ut nostrum ducimus occaecati veniam ut exercitationem voluptatibus.
                </p>
              </div>
            </div>
            <div
              class="MuiPaper-root MuiCard-root makeStyles-root-4 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                class="MuiCardHeader-root"
              >
                <div
                  class="MuiCardHeader-avatar"
                >
                  <div
                    aria-label="recipe"
                    class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-7"
                  >
                    <img
                      class="MuiAvatar-img"
                      srcset="data:image/svg+xml;charset=UTF-8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20version%3D%221.1%22%20baseProfile%3D%22full%22%20width%3D%22undefined%22%20height%3D%22undefined%22%3E%20%3Crect%20width%3D%22100%25%22%20height%3D%22100%25%22%20fill%3D%22grey%22%2F%3E%20%20%3Ctext%20x%3D%220%22%20y%3D%2220%22%20font-size%3D%2220%22%20text-anchor%3D%22start%22%20fill%3D%22white%22%3Eundefinedxundefined%3C%2Ftext%3E%20%3C%2Fsvg%3E"
                    />
                  </div>
                </div>
                <div
                  class="MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
                  >
                    Dr. Mrs. Reanna Walter
                  </span>
                  <span
                    class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                  >
                    Senior Mobility Architect
                  </span>
                </div>
              </div>
              <div
                class="MuiCardContent-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Bogan Group
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                >
                  Molestias ab esse impedit aliquam quo vel dolorem.
Dolor consequatur quis nobis nemo asperiores nam ea aut.
Nam corrupti necessitatibus quas.
Ea odio sit odio quae voluptas id error.
                </p>
              </div>
            </div>
            <div
              class="jsx-2990347280 hiddenItem"
            />
            <div
              class="jsx-2990347280 hiddenItem"
            />
            <div
              class="jsx-2990347280 hiddenItem"
            />
          </div>
          <div
            class="jsx-2990347280 loader"
          >
            Loading ...
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
